src/app/modules/lti/pages/lti/lti.component.ts

```diff
@@ -0,0 +1,105 @@
+import { ItemRouter } from 'src/app/shared/routing/item-router';
+import { LayoutService } from 'src/app/shared/services/layout.service';
+
+const explicitEntryWithNoResultError = new Error('no result');
+const noChildError = new Error('no first child');
+const noContentIdError = new Error('no content id');
+type LTIError = 'no content id' | 'fetch error' | 'explicit entry with no result' | 'no child';
+
+@Component({
+  selector: 'alg-lti',
+  templateUrl: './lti.component.html',
+  styleUrls: [ './lti.component.scss' ],
+})
+export class LTIComponent {
+
+  readonly navigationData$ = this.activatedRoute.queryParamMap.pipe(
+    switchMap(queryParams => {
+      const contentId = queryParams.get('content_id');
+      if (!contentId) return of(errorState(noContentIdError));
+      return this.getNavigationData(contentId).pipe(mapToFetchState());
+    }),
+  );
+  readonly error$ = this.navigationData$.pipe(
+    map((state): LTIError | undefined => {
+      if (!state.isError) return undefined;
+      switch (state.error) {
+        case noContentIdError: return 'no content id';
+        case noChildError: return 'no child';
+        case explicitEntryWithNoResultError: return 'explicit entry with no result';
+        default: return 'fetch error';
+      }
+    })
+  );
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private itemRouter: ItemRouter,
+    private layoutService: LayoutService,
+    private getItemByIdService: GetItemByIdService,
+    private getResultsService: GetResultsService,
+    private getItemPathService: GetItemPathService,
+    private resultActionsService: ResultActionsService,
+    private getItemChildrenService: GetItemChildrenService,
+  ) {
+    this.layoutService.toggleTopRightControls(false);
+    this.layoutService.toggleFullFrameContent(true, false);
+
+    this.navigationData$.subscribe({
+      next: state => {
+        if (!state.isReady) return;
+        const { firstChild, path, result } = state.data;
+        const itemRoute = fullItemRoute('activity', firstChild.id, [ ...path, state.data.itemId ], { attemptId: result.attemptId });
+        this.itemRouter.navigateTo(itemRoute, { navExtras: { replaceUrl: true } });
+      },
+    });
+  }
+
+  private getNavigationData(itemId: string): Observable<{ itemId: string, firstChild: ItemChild, path: string[], result: Result }> {
+    const item$ = this.getItemByIdService.get(itemId).pipe(shareReplay(1));
+    const path$ = this.getItemPathService.getItemPath(itemId).pipe(shareReplay(1));
+    const results$ = this.getResultsService.get(itemId, { attempt_id: '0' });
+
+    const bestResult$ = forkJoin([ item$, results$ ]).pipe(
+      switchMap(([ item, results ]) => {
+        if (results.length === 0 && !implicitResultStart(item)) throw explicitEntryWithNoResultError;
+        return path$.pipe(switchMap(path => this.getBestResultOrStartOne(item, results, path)));
+      }),
+      shareReplay(1),
```

Comment:
You are taking a wrong direction on the results/attempts. You assume that your can rely on the parent attempt "0" but on the results you search for the best result. Let's make it simpler.
As you don't know you parent attempt, no need for all this, just `startWithoutAttempt`  when you know you item path. You should get an attempt that you can use.
Also, you probably don't need to fetch the item info, if "startWithoutAttempt" fails, that's probably because you or one of your ancestors is implicit entry, no need for extra checks.
It is not clear to me at this state (I have read the rest of the code yet) what you need ... but I assume that with these two calls, you'll get the item path and the attempt id, that's enough to fetch the children.
That should simplify a lot your code.

Reply:
Fixed
path-from-root fails before start-without-attempt, I'll handle it this way